### PR TITLE
Squid Auth maximum user IP configuration. Implements #10725

### PR DESCRIPTION
**Pull request by vktg**
- [X] Redmine Issue: https://redmine.pfsense.org/issues/10725
- [X] Ready for review

Allows to set `max_user_ip` in the WebGUI

from http://www.squid-cache.org/Doc/config/acl/:
```
acl aclname max_user_ip [-s] number
# This will be matched when the user attempts to log in from more
# than <number> different ip addresses. The authenticate_ip_ttl
# parameter controls the timeout on the ip entries. [fast]
# If -s is specified the limit is strict, denying browsing
# from any further IP addresses until the ttl has expired. Without
# -s Squid will just annoy the user by "randomly" denying requests.
# (the counter is reset each time the limit is reached and a
# request is denied)
# NOTE: in acceleration mode or where there is mesh of child proxies,
# clients may appear to come from multiple addresses if they are
# going through proxy farms, so a limit of 1 may cause user problems.
```

### CODE REVIEW

File: www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml [hunk 1]
```diff
@@ -141,6 +141,22 @@
 			<size>5</size>
 			<default_value>5</default_value>
 		</field>
+		<field>
+			<fielddescr>Authentication Max User IP</fielddescr>
```
Comment (jim-p) [inline www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml:145]: Change "IP" to "IP Addresses"

File: www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml [hunk 2]
```diff
@@ -141,6 +141,22 @@
...
+			<fieldname>max_user_ip</fieldname>
+			<description>
+				<![CDATA[
+				This will be matched when the user attempts to log in from more than <number> different IP addresses.<br/>
```
Comment (jim-p) [inline www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml:149]: "Enforces a limit to the number of unique IP addresses from which a single user can login."

File: www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml [hunk 3]
```diff
@@ -141,6 +141,22 @@
...
+				Browsing from any further IP addresses is denying until the Authentication TTL has expired.<br/>
```
Comment (jim-p) [inline www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml:150]: "Attempts to login from additional IP addresses are denied until the Authentication TTL has expired"

File: www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml [hunk 4]
```diff
@@ -141,6 +141,22 @@
...
+				<span class="text-info">Default: none</span>
+				<div class="infoblock">
+				<span class="text-info">Note:</span> In acceleration mode or where there is mesh of child proxies, clients may appear to come from multiple addresses if they are going through proxy farms, so a limit of 1 may cause user problems.
```
Comment (jim-p) [inline www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml:153]: This seems a bit confusing in the wording. Perhaps:

> Acceleration mode or multiple child proxies in a mesh or farm may cause clients to appear as though they are using multiple IP addresses. A limit of 1 may cause problems for users in these cases.

Comment (vktg) [reply www/pfSense-pkg-squid/files/usr/local/pkg/squid_auth.xml]: Ok, fixed

**Pull request merged**